### PR TITLE
pod pkg: added bond annotation func network.go

### PR DESCRIPTION
**Pull request by gkopels**
Add staticAnnotation func with name only to pod > network.go

### CODE REVIEW

File: pkg/pod/network.go [hunk 1]
```diff
@@ -58,3 +65,20 @@ func StaticIPAnnotationWithInterfaceMacAndNamespace(
 
 	return baseAnnotation
 }
+
+// StaticIPBondAnnotationWithInterface.
```
Comment (evgenLevin) [inline pkg/pod/network.go:69]: fix

File: pkg/pod/network.go [hunk 2]
```diff
@@ -58,3 +65,20 @@ func StaticIPAnnotationWithInterfaceMacAndNamespace(
...
+		annotation = append(annotation, StaticAnnotation(sriovNetName))
+	}
+
+	bond := StaticAnnotation(bondNadName)
+	bond.InterfaceRequest = bondIntName
+	bond.IPRequest = ipAddrBond
+
+	annotation = append(annotation, bond)
+
+	return annotation
```
Comment (evgenLevin) [inline pkg/pod/network.go:83]: return  append(annotation, bond)

File: pkg/pod/network.go [hunk 3]
```diff
@@ -58,3 +65,20 @@ func StaticIPAnnotationWithInterfaceMacAndNamespace(
 
 	return baseAnnotation
 }
+
+// StaticIPBondAnnotationWithInterface.
+func StaticIPBondAnnotation(bondNadName, bondIntName string, sriovNetworkNameList,
```
Comment (evgenLevin) [inline pkg/pod/network.go:70]:
```
func StaticIPBondAnnotation(
     bondNadName, bondIntName string, sriovNetworkNameList,ipAddrBond []string) []*multus.NetworkSelectionElement
```

Comment (evgenLevin) [inline pkg/pod/network.go:70]: StaticIPBondAnnotationWithInterface

File: pkg/pod/network.go (hunk 1 above)
Comment (gkopels) [reply pkg/pod/network.go:69]: fixed

File: pkg/pod/network.go (hunk 2 above)
Comment (gkopels) [reply pkg/pod/network.go:83]: changed

File: pkg/pod/network.go (hunk 3 above)
Comment (gkopels) [reply pkg/pod/network.go:70]: changed

Comment (gkopels) [reply pkg/pod/network.go:70]: changed

**Comment by evgenLevin** [comment]
lgtm

**Review by kononovn — CHANGES_REQUESTED** [review]
Please check comment above

File: pkg/pod/network.go [hunk 4]
```diff
@@ -58,3 +65,19 @@ func StaticIPAnnotationWithInterfaceMacAndNamespace(
...
+// StaticIPBondAnnotationWithInterface defines static name for bonded interfaces and name, interface and IP for the
+// main bond interface.
+func StaticIPBondAnnotationWithInterface(
+	bondNadName, bondIntName string, sriovNetworkNameList, ipAddrBond []string) []*multus.NetworkSelectionElement {
+	annotation := []*multus.NetworkSelectionElement{}
+	for _, sriovNetName := range sriovNetworkNameList {
+		annotation = append(annotation, StaticAnnotation(sriovNetName))
+	}
+
+	bond := StaticAnnotation(bondNadName)
```
[the remark this answers is not in the document]
Comment (gkopels) [reply pkg/pod/network.go:78]: Because two of the interfaces dont have IP addresses.

File: pkg/pod/network.go [hunk 5]
```diff
@@ -58,3 +65,18 @@ func StaticIPAnnotationWithInterfaceMacAndNamespace(
 
 	return baseAnnotation
 }
+
+// StaticIPBondAnnotationWithInterface defines static name for bonded interfaces and name, interface and IP for the
+// main bond int.
+func StaticIPBondAnnotationWithInterface(
+	bondNadName, bondIntName string, sriovNetworkNameList, ipAddrBond []string) []*multus.NetworkSelectionElement {
```
Comment (kononovn) [inline pkg/pod/network.go:72]: This function has to return just bond annotation and not list of annotations:
*multus.NetworkSelectionElement

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**